src/server/scripts/Kalimdor/DireMaul/instance_dire_maul.cpp

```diff
@@ -60,8 +78,203 @@ class instance_dire_maul : public InstanceMapScript
+                    _crystalGUIDs[3] = go->GetGUID();
+                    break;
+                case GO_CRYSTAL_05:
+                    _crystalGUIDs[4] = go->GetGUID();
+                    break;
+                case GO_FORCEFIELD:
+                    _forcefieldGUID = go->GetGUID();
+                    if (GetBossState(DATA_FORCEFIELD) != DONE)
+                        _events.ScheduleEvent(EVENT_CRYSTAL_CREATURE_STORE, 1s);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        ObjectGuid GetGuidData(uint32 type) const override
+        {
+            switch (type)
+            {
+                case GO_CRYSTAL_01:
+                    return _crystalGUIDs[0];
+                case GO_CRYSTAL_02:
+                    return _crystalGUIDs[1];
+                case GO_CRYSTAL_03:
+                    return _crystalGUIDs[2];
+                case GO_CRYSTAL_04:
+                    return _crystalGUIDs[3];
+                case GO_CRYSTAL_05:
+                    return _crystalGUIDs[4];
+                case GO_FORCEFIELD:
+                    return _forcefieldGUID;
+                case NPC_IMMOLTHAR:
+                    return _immoGUID;
+                default:
+                    break;
+            }
+            return ObjectGuid::Empty;
+        }
+
+        void Update(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_CRYSTAL_CREATURE_STORE:
+                        Crystalcreaturestore();
+                        _events.ScheduleEvent(EVENT_CRYSTAL_CREATURE_CHECK, 3s);
+                        break;
+                    case EVENT_CRYSTAL_CREATURE_CHECK:
+                        Crystalcreaturecheck();
+                        if ((GetBossState(DATA_FORCEFIELD) != DONE))
+                            _events.ScheduleEvent(EVENT_CRYSTAL_CREATURE_CHECK, 3s);
+                        break;
+                    default:
+                         break;
+                }
+            }
+        }
+
+        void Crystalcreaturestore()
+        {
+            for (uint8 i = 0; i < 5; ++i) // we store creatures in a list for all 5 crystals
+            {
+                uint8 creatureCount = 0;
+
+                if (GameObject* crystal = instance->GetGameObject(_crystalGUIDs[i]))
+                {
+                    for (uint8 j = 0; j < 2; ++j)  // once per creature type from CrystalMobs
+                    {
+                        std::list<Creature*> creatureList;
+                        GetCreatureListWithEntryInGrid(creatureList, crystal, CrystalMobs[j], 30.0f);
+                        for (std::list<Creature*>::iterator itr = creatureList.begin(); itr != creatureList.end(); ++itr)
+                        {
+                            if (Creature* creature = *itr)
```

Comment:
can this ever be null ? it sounds strange for that to happen. You can probably just
```cpp
(*itr)->GetGUID();
```